src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/AppealServiceTest.kt

```diff
@@ -279,5 +346,33 @@ class AppealServiceTest {
         this.application == application &&
         this.assessment == assessment &&
         this.createdBy == createdByUser
+
+    private fun DomainEvent<AssessmentAppealedEnvelope>.matches() =
+      this.applicationId == application.id &&
+        this.assessmentId == null &&
+        this.bookingId == null &&
+        this.crn == application.crn &&
+        withinSeconds(10).matches(this.occurredAt.toString()) &&
+        this.data.matches()
+
+    private fun AssessmentAppealedEnvelope.matches() =
+      this.eventDetails.applicationId == application.id &&
+        this.eventDetails.applicationUrl == "http://frontend/applications/${application.id}" &&
```

Comment:
Especially as they are new to our domain event "envelopes", do you think it's also worth describing:

- `appealId`
- `appealUrl`

Reply:
so, perhaps this is nudging us to use an explicit UUID e.g.

```kt
every { applicationAppealUrlTemplate.resolve(any()) } returns
   "http://frontent/applications/${application.id}/appeals/${explicitAppealId)}"
```

Reply:
I've now made this change.